### PR TITLE
[WIP][Unittest] Tensorflow unittest with MNIST

### LINKED ISSUE #994: [Unittest] how to pass test cases for the not available framework
I made a unit test case for tensorflow with MNIST model.
However, it occurs an error because some of the architectures are not able to use tensorflow (like i586).

The problem is, because ssat run whole of test cases in the `tests` directory, we cannot adjust according to the validation of framework.
Now I'm trying to avoid this problem by checking the architecture or the validation of tensorflow at the `runTest.sh` before running testcases.

Are there any better Ideas?

### PR DESCRIPTION
**Pull request by helloahn**
the basic unittest of tensorflow with MNIST model.
it resolves #994 

Signed-off-by: HyoungjooAhn <hello.ahnn@gmail.com>

**Self evaluation:**
1. Build test: [*]Passed [ ]Failed [ ]Skipped
2. Run test: [*]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: tests/nnstreamer_filter_tensorflow/runTest.sh [hunk 1]
```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env bash
+##
+## @file runTest.sh
+## @author HyoungJoo Ahn <hello.ahn@samsung.com>
+## @date Dec 17 2018
+## @brief SSAT Test Cases for NNStreamer
+##
+
```
Comment (myungjoo) [inline tests/nnstreamer_filter_tensorflow/runTest.sh]: Would it be recommended to add what https://github.com/nnsuite/nnstreamer/pull/992 has done here?
@wooksong

Comment (leemgs) [reply tests/nnstreamer_filter_tensorflow/runTest.sh]: @helloahn Please do not use '/'.

**Comment by helloahn** [comment]
Currently, it finds `/usr/lib` or `/usr/local/lib` to check the pkgconfig(.pc) file at the environment.
It's just passed test case when there is no `.pc` file of tensorflow/protobuf.
Are there any better idea?

```bash
[  160s] ==================================================
[  160s]     Test Group nnstreamer_filter_tensorflow Starts.
[  160s] ==================================================
[  160s] [PASSED] Test Group nnstreamer_filter_tensorflow Passed
[  160s] ==================================================
```

File: tests/nnstreamer_filter_tensorflow/runTest.sh [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+##
+TF_OK=0
+PB_OK=0
+
+OUTPUT="$(find /usr/lib -name tensorflow.pc 2>&1 | grep -v 'Permission denied')"
+LENGTH=${#OUTPUT}
+if (( $LENGTH <= 0 ))
+then
+    OUTPUT="$(find /usr/local/lib -name tensorflow.pc 2>&1 | grep -v 'Permission denied')"
+    LENGTH=${#OUTPUT}
```
Comment (myungjoo) [inline tests/nnstreamer_filter_tensorflow/runTest.sh]: As @wooksong has recommended yesterday,
why don't you try to link the tensorflow-filter library with ```ldconfig -l```?

**Comment by myungjoo** [comment]
Users will usually install the binaries only without devel packages. Thus, we cannot expect to have .pc in the target devices. As @wooksong has recommended yesterday, probing with ldconfig -l is much better.

File: tests/nnstreamer_filter_tensorflow/runTest.sh (hunk 2 above)
Comment (helloahn) [reply tests/nnstreamer_filter_tensorflow/runTest.sh]: using `ldconfig -l` result is applied :)

File: tests/nnstreamer_filter_tensorflow/runTest.sh [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	printf "${Blue}Independent Mode${NC}
+"
+fi
+
+# This is compatible with SSAT (https://github.com/myungjoo/SSAT)
+testInit $1
+
+PATH_TO_SO="$(rpm --eval '%{buildroot}%{_libdir}')/libtensor_filter_tfcore.so"
+echo $PATH_TO_SO
+OUTPUT="$(ldconfig -l $PATH_TO_SO 2>&1)"
```
Comment (myungjoo) [inline tests/nnstreamer_filter_tensorflow/runTest.sh]: Checking with full path, hardcoded for build script is not good.

Comment (helloahn) [reply tests/nnstreamer_filter_tensorflow/runTest.sh]: you mean this part `/libtensor_filter_tfcore.so`?

Comment (myungjoo) [reply tests/nnstreamer_filter_tensorflow/runTest.sh]: Other people may need to run SSAT with installed nnstreamer pacakges, not within gbs or pdebuild.

Comment (myungjoo) [reply tests/nnstreamer_filter_tensorflow/runTest.sh]: > you mean this part /libtensor_filter_tfcore.so?

Yes.

Comment (myungjoo) [reply tests/nnstreamer_filter_tensorflow/runTest.sh]: Hints: https://serverfault.com/questions/54736/how-to-check-if-a-library-is-installed

Comment (helloahn) [reply tests/nnstreamer_filter_tensorflow/runTest.sh]: > Other people may need to run SSAT with installed nnstreamer pacakges, not within gbs or pdebuild.

you're right.. I've been concentrated on only CI-pass too much. I need to concern about it.

> Hints: https://serverfault.com/questions/54736/how-to-check-if-a-library-is-installed

I already tried this way, but I can't get the result with `ldconfig -p` in my local environment.
Furthermore, `ldconfig -v | grep SOMETHING` is not working, especially `grep` is not working.

**Comment by helloahn** [comment]
I'll make it with `unittest_sink` because of below reasons.
1. `..tfcore.so` file has no common path: it could be installed at the custom path. In this case, we cannot get `.so`file with `ldconfig`
2. checking the existence and version of packages (tensorflow, protobuf). `dpkg -l` or `find package.pc` is not a common method for every architecture and environment.

My plan is, adding the test of tensorflow at 'unittest_sink', because it already contain a streaming test.
It could be simple because we can get `ENABLE_TENSORFLOW` value easily.

**Pull request closed without merging**